controllers/buildpipeline/buildpipeline_adapter.go

```diff
@@ -262,7 +263,11 @@ func (a *Adapter) getSucceededBuildPipelineRunsForComponent(component *applicati
 
 func (a *Adapter) annotateBuildPipelineRunWithSnapshot(pipelineRun *tektonv1beta1.PipelineRun, snapshot *applicationapiv1alpha1.Snapshot) (*tektonv1beta1.PipelineRun, error) {
 	patch := client.MergeFrom(pipelineRun.DeepCopy())
-	h.AddAnnotation(&pipelineRun.ObjectMeta, tekton.SnapshotNameLabel, snapshot.Name)
+	temp_annotations := map[string]string{}
```

Comment:
nit: `temp_annotations` would probably be easier to understand as something like `newAnnotations`.